include/rmm/device_scalar.hpp

```diff
@@ -92,6 +146,37 @@ class device_scalar {
 
  private:
   rmm::device_buffer buff{sizeof(T)};
+
+  template<bool synchronize>
+  inline T _value(cudaStream_t stream) const {
+    T host_value{};
+    _memcpy<synchronize>(&host_value, buff.data(), sizeof(T), stream);
+    return host_value;
+  }
+
+  template<bool synchronize>
+  inline void _set_value(T host_value, cudaStream_t stream) {
+    _memcpy<synchronize>(buff.data(), &host_value, sizeof(T), stream);
+  }
+
+  template<bool synchronize>
+  inline void _memcpy(void *dst, const void *src, size_t count, cudaStream_t stream) const {
+    auto status = cudaMemcpyAsync(dst, src, count, cudaMemcpyDefault, stream);
+
+    if (cudaSuccess != status) {
+      throw std::runtime_error{"Device memcpy failed."};
+    }
+
+    if (false == synchronize) {
```

Comment:
Prefer positive semantics:
```suggestion
    if (synchronize) {
     if (cudaSuccess != cudaStreamSynchronize(stream)) {
       throw std::runtime_error{"Stream sync failed."};
     }
   }
```